src/main/java/com/iexec/core/task/update/TaskUpdateManager.java

```diff
@@ -73,14 +76,14 @@ public TaskUpdateManager(TaskService taskService,
                              ReplicatesService replicatesService,
                              ApplicationEventPublisher applicationEventPublisher,
                              WorkerService workerService,
-                             BlockchainAdapterService blockchainAdapterService,
+                             BlockchainAdapterApiClient blockchainAdapterApiClient,
                              SmsService smsService) {
         this.taskService = taskService;
         this.iexecHubService = iexecHubService;
         this.replicatesService = replicatesService;
         this.applicationEventPublisher = applicationEventPublisher;
         this.workerService = workerService;
-        this.blockchainAdapterService = blockchainAdapterService;
+        this.blockchainAdapterService = new BlockchainAdapterService(blockchainAdapterApiClient, WATCH_PERIOD_SECONDS, MAX_ATTEMPTS);
```

Comment:
Why don't you inject the bean instead of creating it here?

Reply:
It is the only place where it is used and it allowed me to pass the 80% coverage threshold.
I don't really know the best solution between single use spring beans or traditional java object instances creation for this kind of situation.

Reply:
reverted in last commits